datafusion/expr/src/expr_fn.rs

```diff
@@ -1073,6 +1075,34 @@ pub fn create_udaf(
     ))
 }
 
+/// Creates a new UDAF with a specific signature, state type and return type.
```

Comment:
Adding this API effectively means that trait based user defined aggregates (e.g. `AggregateUDFImpl`) can't take advantage of sorting.

I think the hope is to move all aggregate functions to use the new trait based API eventually, so using a new function seems to be at odds with that goal

https://github.com/apache/arrow-datafusion/blob/edec4189242ab07ac65967490537d77e776aad5c/datafusion/expr/src/udaf.rs#L242

Reply:
What would you think about changing the `AggregateUDFImpl`

```rust
impl AggregateUDFImpl { 
...
    fn accumulator(&self, arg: &DataType) -> Result<Box<dyn Accumulator>>;
...
}
```

To something like the following

```rust
impl AggregateUDFImpl { 
...
    fn accumulator(&self, arg: &DataType, sort_exprs: Vec<Expr>) -> Result<Box<dyn Accumulator>>;
...
}
```

It would be an API change though (but this API is pretty new).

Reply:
Do we need schema or just sort_exprs?

1. fn accumulator(&self, arg: &DataType, sort_exprs: Vec<Expr>, schema: Option<Schema>)

Reply:
Passing in the schema is a good call